src/components/Inputs/TaxIdInput/TaxIdInput.vue

```diff
@@ -0,0 +1,210 @@
+  },
+
+  data() {
+    return {
+      state: {
+        isValidDocument: true,
+        message: '',
+        inputValue: '',
+        disabled: false,
+      },
+    };
+  },
+
+  computed: {
+    taxIdInputClass() {
+      return !this.state.isValidDocument ? 'error' : '';
+    },
+
+    taxIdInputStyle() {
+      return {
+        background: `var(--color-${this.background})`,
+        border: `1px solid var(--color-${
+          this.state.isValidDocument ? this.color : 'danger'
+        })`,
+      };
+    },
+
+    taxIdInputPlaceholder() {
+      if (!this.placeholder && this.allowedDocuments.length === 1)
+        return documentTypesData[this.allowedDocuments[0]].placeholder;
+
+      return this.placeholder;
+    },
+
+    maxLength() {
+      const maxLength = this.allowedDocuments.map(
+        documentType => documentTypesData[documentType].formattedLength,
+      );
+
+      return Math.max(...maxLength);
+    },
+
+    taxIdInput: {
+      get() {
+        return this.state.inputValue.length <= 11
+          ? this.stringToCpfFormat(this.state.inputValue)
+          : this.stringToCnpjFormat(this.state.inputValue);
+      },
+
+      set(document) {
+        const formatDocumentValue = document
+          .replace(/[^\w\s]/gi, '')
+          .replace(/\s+/g, '')
+          .trim();
+        this.$emit('input', formatDocumentValue);
+      },
+    },
+  },
+
+  watch: {
+    value() {
+      this.state.inputValue = String(this.value);
+    },
+    allowedDocuments() {
+      const hasAllowedDocuments = this.allowedDocuments.length >= 1;
+
+      this.updateValidationField({ message: !hasAllowedDocuments ? 'Informe um tipo de documento. Consulte a documentação do componente!' : '' });
+
+      this.state.disabled = !hasAllowedDocuments;
+    },
+  },
+
+  mounted() {
+    if (this.taxIdInput) this.validateDocument();
+  },
+
+  methods: {
+    stringToCnpjFormat(cnpj) {
+      cnpj = cnpj.replace(/\D/g, '');
+      cnpj = cnpj.replace(/^(\d{2})(\d)/, '$1.$2');
+      cnpj = cnpj.replace(/^(\d{2})\.(\d{3})(\d)/, '$1.$2.$3');
+      cnpj = cnpj.replace(/\.(\d{3})(\d)/, '.$1/$2');
+      cnpj = cnpj.replace(/(\d{4})(\d)/, '$1-$2');
+      return cnpj;
+    },
+
+    stringToCpfFormat(cpf) {
+      cpf = cpf.replace(/\D/g, '');
+      cpf = cpf.replace(/(\d{3})(\d)/, '$1.$2');
+      cpf = cpf.replace(/(\d{3})(\d)/, '$1.$2');
+      cpf = cpf.replace(/(\d{3})(\d{1,2})$/, '$1-$2');
+      return cpf;
+    },
+
+    validateLength(documentToValidate) {
+      // eslint-disable-next-line max-len
+      const hasValidLength = this.allowedDocuments.some(
+        documentType => documentToValidate.length === documentTypesData[documentType].length,
+      );
+
+      return hasValidLength;
+    },
+
+    validateDocument() {
+      const documentToValidate = this.taxIdInput
+        .replace(/[^\w\s]/gi, '')
+        .replace(/\s+/g, '')
+        .trim();
+
+      if (!this.required && !documentToValidate.length) return;
```

Comment:
ta correto? se for required e não tiver length ele vai passar por aqui?

Reply:
sim! pois quando o campo não for obrigatório e o usuário não ter informado nenhum valor, não há nenhuma requisição para ser feita.